README.md

```diff
@@ -72,6 +72,27 @@ For example:
 # Hello {props.name}
 ```
 
+#### JSDoc in MDX with Prettier
+
+If you’re using Prettier, you’ll need to ignore any JSDoc comments until it is
+supported (issues [MDX 3](https://github.com/prettier/prettier/issues/16457)
+and
+[MDX v2+ JSDoc comments](https://github.com/prettier/prettier/issues/16457)):
+
+```mdx
+{/* prettier-ignore-start */}
```

Comment:
I don’t think using Prettier ignore comments is a good solution. Prettier doesn’t properly support MDX 3 yet overall. If you use this comment, another issue will pop up elsewhere. I would rather add the recommendation to add `.mdx` to `.prettierignore`.

Reply:
hmm, I'm still going to try it out and stress test it for a while - I didn't see any other problems on a long document

what about:

1. "since MDX 3 is not officially supported yet, we recommend ignoring `*.mdx`: <example>"
2. "but another option is to ignore the parts that are not yet working: <existing example in PR>"

I think there will be at least some users who will want to stick with Prettier for most things, and ignore the parts that don't work, including these JSDoc comments

Reply:
Prettier actively breaks good looking proper MDX (2+) files, not just expressions. MDX 2 got a ton of power and Prettier prevents people from using it.
I personally really really really would not recommend it, but I’ll leave this repo here to Remco.

Reply:
I want this to be a small section explaining Prettier has issues, like the ESLint section. I don’t want to go into depth too much about which parts work or don’t work with Prettier (today). Something like:

````markdown
## Prettier

Prettier supports MDX 1, but MDX 2 and 3 are [not officially supported](https://github.com/prettier/prettier/issues/12209) yet.
We recommend against formatting MDX files with Prettier.
To opt out, add the following line to your `.prettierignore` file:

```ignore
*.mdx
```
````

Reply:
Ok, sounds mostly good

what do you think about my recommendation to add a second part after the codeblock with `{/* prettier-ignore-start */}`?

> 2\. "but another option is to ignore the parts that are not yet working: <existing example in PR>"

my concern / reasoning for this is that I think it's not a very well-known option for users, and I think some users would rather ignore and still get some benefit from Prettier (eg. I like the formatting that it does apply to the other, supported sections)

Reply:
As long as Prettier doesn’t properly support MDX 2/3, I really don’t think the MDX sources should recommend Prettier as a formatter. Of course you are free to have your own opinions and recommendations, but as the MDX team we should play on the safe side.

I think of this the same as using Prettier to support SVG/XML files by configuring Prettier to treat them as HTML. Sure, it’s largely compatible, but they’re really not the same. You will run into unexpected issues.